tsk/fs/ntfs_dent.cpp

```diff
@@ -514,7 +514,12 @@ ntfs_proc_idxentry(NTFS_INFO * a_ntfs, TSK_FS_DIR * a_fs_dir,
                 continue;
             }
         }
-
+	// WARNING: tsk_UTF16toUTF8 used in ntfs_dent_copy_short_only cannot check for
+	// all out of bounds cases.
+        if ((uintptr_t) a_idxe + idxlen > endaddr_alloc) {
+            a_idxe = (ntfs_idxentry *) ((uintptr_t) a_idxe + 4);
+            continue;
```

Comment:
Shouldn't we actually fix this:
https://github.com/sleuthkit/sleuthkit/blob/e119503b6fd4472dab1ce8209f770ac1aafa9b19/tsk/base/tsk_unicode.c#L155
and make it `while (source+1 < sourceEnd) {` ?

Reply:
Yes. If we change `source` to `source+1` in `tsk_UTF16toUTF8`, I believe it solves this problem. What do you think?

<img width="757" alt="image" src="https://github.com/user-attachments/assets/e7ec4a12-a81f-46da-99c8-30e97f87b28c" />

Reply:
Ah, OOB means `out of bounds`.

Reply:
Actually, I can probably keep this as well.